src/utils/response/resourceResponse.ts

```diff
@@ -9,7 +9,7 @@ export type ResourceResponse<T> = {
   success: boolean;
   results: T[];
   deleted: DeletedEntry[];
-  affected?: Record<string, any[]>;
+  affected: Record<string, any[]>; // TODO this should not be an record, it is should be AffectedBy<T>
```

Comment:
Just on a side note: `any` isn't allowed with updated ESLint rules, there will be a bigger refactoring of response types to get that in proper shape: https://github.com/TimeTac/js-client-library/pull/82/files#diff-9b27af540065038ee20390203cc238464beb9804ee7974ebf3ee307894e817ea
Also `DeltaSyncResponse` is interesting I think. Affected is probably still wrong in that PR, I think it's missing an Array. We can talk about it also in the meeting with @whosshell

Reply:
FYI: affected is returned as an object by the backed, similar to DeltaSyncResponse.

Reply:
Yes, probably the type would be 
```
  type Affected<R extends Resource> = {
    [r in R]: Entity<r>[]
  }

  type Resource = keyof Resources & string;
  type Entity<R extends Resource> = Resources[R];
  type Resources = {
    timeTrackings: TimeTracking;
    users: User,
  }
```